app/services/spree_stripe/create_source.rb

```diff
@@ -1,14 +1,15 @@
 module SpreeStripe
   class CreateSource
-    def initialize(stripe_charge:, order:, gateway:)
-      @order = order
-      @stripe_charge = stripe_charge
+    def initialize(payment_method_details:, payment_method_id:, gateway:, billing_details:, order: nil, user: nil)
```

Comment:
```suggestion
    def initialize(stripe_payment_method_details:, stripe_payment_method_id:, gateway:, stripe_billing_details:, order: nil, user: nil)
```

so we can distinguish what is a s spree object and what is information from stripe APIs